src/com/projectkorra/projectkorra/GeneralMethods.java

```diff
@@ -1064,7 +840,7 @@ public static long getGlobalCooldown() {
 	public static double getHorizontalDistance(final Location one, final Location two) {
 		final double x = one.getX() - two.getX();
 		final double z = one.getZ() - two.getZ();
-		return Math.sqrt((x * x) + (z * z));
+		return FastMath.sqrt((x * x) + (z * z));

```

Comment:
Please see https://www.element84.com/blog/improving-java-math-performance-with-jafama for speed comparasions.

Yes, I know that in the grand scheme of things, micro optimizations won't help a lot. But that isn't the point. This is technically faster, and if it's available, there is no reason to not use these. The inaccuracy of FastMath compared to Math will only really be seen in the 5th decimal place or further. Meaning that for what we use, it is impossible to see a difference (as you'd need something bigger than 10,000 blocks for it to be a block out of where it should be)

Reply:
I'm not sure what all he did, but Vahagn did some benchmarks on FastMath compared to Math and it seemed to be slower for his tests.

Reply:
Yes I looked into that link provided, however I decided to run my own benchmarks reflecting the typical PK use case as it bothered my intuition. My own benchmarks contradict those that were done in that web blog. 
```

public class MathBenchmarkDriver {

	public static void main(String[] args) {
		long a = 0;
		long b = 0;
		
		int n = 360;

		long startTime_1 = 0;
		long endTime_1 = 0;
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_1 = System.nanoTime();
			FastMath.sin(in);
			endTime_1 = System.nanoTime();
			a += (endTime_1 - startTime_1);
		}

		
		long startTime_2 = 0;
		long endTime_2 = 0;
		
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_2 = System.nanoTime();
			Math.sin(in);
			endTime_2 = System.nanoTime();
			b += (endTime_2 - startTime_2);
		}
		
		double avg_a = a / n;
		double avg_b = b / n;
		
		System.out.println("Sin Math: " + avg_b);
		System.out.println("Sin FastMath: " + avg_a);
		System.out.println("Sin Diff: " + (avg_b - avg_a));
		
		a = 0;
		b = 0;
		
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_1 = System.nanoTime();
			FastMath.cos(in);
			endTime_1 = System.nanoTime();
			a += (endTime_1 - startTime_1);
		}
		
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_2 = System.nanoTime();
			Math.cos(in);
			endTime_2 = System.nanoTime();
			b += (endTime_2 - startTime_2);
		}
		
		avg_a = a / n;
		avg_b = b / n;
		
		System.out.println("Cos Math: " + avg_b);
		System.out.println("Cos FastMath: " + avg_a);
		System.out.println("Cos Diff: " + (avg_b - avg_a));
		
		a = 0;
		b = 0;

		for(int i = 0; i < n; i++) {
			double in = Math.random() * n;
			startTime_1 = System.nanoTime();
			FastMath.sqrt(in);
			endTime_1 = System.nanoTime();
			a += (endTime_1 - startTime_1);
		}
		
		for(int i = 0; i < n; i++) {
			double in = Math.random() * n;
			startTime_2 = System.nanoTime();
			Math.sqrt(in);
			endTime_2 = System.nanoTime();
			b += (endTime_2 - startTime_2);
		}
		
		avg_a = a / n;
		avg_b = b / n;
		
		System.out.println("Sqrt Math: " + avg_b);
		System.out.println("Sqrt FastMath: " + avg_a);
		System.out.println("Sin Diff: " + (avg_b - avg_a));
		
	}
}

```

For n = 360 (typical PK usecase)

![image](https://user-images.githubusercontent.com/19560788/184525296-e6c73835-91cf-48d9-8467-9fb2ea40f843.png)
![image](https://user-images.githubusercontent.com/19560788/184525298-ce33ef57-efa1-4d18-b560-0b67d7885366.png)


For n = 500000

![image](https://user-images.githubusercontent.com/19560788/184525266-d41a9a81-acee-461a-8787-aba64c7387fe.png)
![image](https://user-images.githubusercontent.com/19560788/184525273-7502fde5-cd76-4468-9c41-3cdf44a30cf5.png)

For n = 1000000

![image](https://user-images.githubusercontent.com/19560788/184525322-8656566d-727b-4993-81e2-8be7fd6d872e.png)
![image](https://user-images.githubusercontent.com/19560788/184525331-0a7a3db1-7c99-453b-9586-7d695fa0bfdf.png)

I dont think the performance hit for n < 500,000 is justifies using it.

Reply:
I doubted the benchmarks on that website when I did my own preliminary search since it violated my intuition.

After running benchmarks using this
```

public class MathBenchmarkDriver {

	public static void main(String[] args) {
		long a = 0;
		long b = 0;
		
		int n = 360;
		
		for(int i = 0; i < 1000000; i++) {
			double in = Math.toRadians(i % 360);
			FastMath.sin(in);
			Math.sin(in);
			FastMath.cos(in);
			Math.cos(in);
			FastMath.sqrt(in);
			Math.sqrt(in);
		}
		
		long startTime_1 = 0;
		long endTime_1 = 0;
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_1 = System.nanoTime();
			FastMath.sin(in);
			endTime_1 = System.nanoTime();
			a += (endTime_1 - startTime_1);
		}

		
		long startTime_2 = 0;
		long endTime_2 = 0;
		
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_2 = System.nanoTime();
			Math.sin(in);
			endTime_2 = System.nanoTime();
			b += (endTime_2 - startTime_2);
		}
		
		double avg_a = a / n;
		double avg_b = b / n;
		
		System.out.println("Sin Math: " + avg_b);
		System.out.println("Sin FastMath: " + avg_a);
		System.out.println("Sin Diff: " + (avg_b - avg_a));
		
		a = 0;
		b = 0;
		
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_1 = System.nanoTime();
			FastMath.cos(in);
			endTime_1 = System.nanoTime();
			a += (endTime_1 - startTime_1);
		}
		
		for(int i = 0; i < n; i++) {
			double in = Math.toRadians(i % 360);
			startTime_2 = System.nanoTime();
			Math.cos(in);
			endTime_2 = System.nanoTime();
			b += (endTime_2 - startTime_2);
		}
		
		avg_a = a / n;
		avg_b = b / n;
		
		System.out.println("Cos Math: " + avg_b);
		System.out.println("Cos FastMath: " + avg_a);
		System.out.println("Cos Diff: " + (avg_b - avg_a));
		
		a = 0;
		b = 0;

		for(int i = 0; i < n; i++) {
			double in = Math.random() * n;
			startTime_1 = System.nanoTime();
			FastMath.sqrt(in);
			endTime_1 = System.nanoTime();
			a += (endTime_1 - startTime_1);
		}
		
		for(int i = 0; i < n; i++) {
			double in = Math.random() * n;
			startTime_2 = System.nanoTime();
			Math.sqrt(in);
			endTime_2 = System.nanoTime();
			b += (endTime_2 - startTime_2);
		}
		
		avg_a = a / n;
		avg_b = b / n;
		
		System.out.println("Sqrt Math: " + avg_b);
		System.out.println("Sqrt FastMath: " + avg_a);
		System.out.println("Sin Diff: " + (avg_b - avg_a));
		
	}
}
```

For n = 500000
![image](https://user-images.githubusercontent.com/19560788/184526151-e132da19-12a3-49a8-aed6-a9d34a9d48d0.png)
![image](https://user-images.githubusercontent.com/19560788/184526156-92f6fb60-d4b4-4e0c-b2b6-427d9a0dfe2b.png)

We see variable but always nearly the same performance between the two.

For n =  360 (typical PK use case)

![image](https://user-images.githubusercontent.com/19560788/184526170-195dfc43-4f7c-4a7c-963a-fb205703eef0.png)
![image](https://user-images.githubusercontent.com/19560788/184526186-d33f3d41-886f-48bd-b01d-f93fd9255677.png)

We see consistently that Math.sin performs better than FastMath.cos. FastMath.cos and Math.cos have variable but equal performance. and Math.sqrt vs FastMath.sqrt is by default going to delegate to Math.sqrt, therefore its always slightly faster due to the additional function call overhead.

With this in mind, I don't see any reason to use this library, because the performance is marginally better if at all, and because its entirely written in Java unlike Math library native code, cannot take advantage of any low level hardware optimizations that hosts often implement to boost performance.

Reply:
so true king don't use FastMath